Supraja work

suprajakosuri01 opened the pull request:
## PR Title
Added GPT Interaction with Diverse Prompt Patterns and Inclusion of External Course Links for Learning about LLMs and Prompting

## Description of Changes
Added GPT Interaction with Diverse Prompt Patterns -[Readme.md](https://github.com/suprajakosuri01/humanitarians-OPT-fork/tree/supraja-work/PROMPT_ENGINEERING_AND_AI/LLM-Intro_Prompting-SlideDecks_Videos_Quizzes#readme)]
Added External Course Links for Learning about LLMs and Prompting-https://docs.google.com/presentation/d/1aIFUSefNdRyMK_i3EkXkH4qNDRn_laGtggvohQ0812I/edit#slide=id.p3


### Sections Changed
In Introduction to LLMs and Prompting_Module1 , google docs added slides 26-29
https://docs.google.com/presentation/d/1aIFUSefNdRyMK_i3EkXkH4qNDRn_laGtggvohQ0812I/edit#slide=id.p3

Added link to view GPT Interaction with Diverse Prompt Patterns in 
https://docs.google.com/document/d/16pKaAONgvsCgLn1GQo1D5ZBB5tCjA9_2ctoJ4Nhqze4/edit

[Readme.md](https://github.com/suprajakosuri01/humanitarians-OPT-fork/tree/supraja-work/PROMPT_ENGINEERING_AND_AI/LLM-Intro_Prompting-SlideDecks_Videos_Quizzes#readme)]


### Screenshots
<img width="1344" alt="Screenshot 2024-02-29 at 5 22 06 PM" src="https://github.com/parthasarathydNU/humanitarians-OPT-fork/assets/98271852/d4467b35-b061-4d8b-bd37-ef703be5ad66">

### Issues Faced
No issues Faced


## Review Request
@parthasarathydNU 

## Checklist Before Submission
- [ ] I have reviewed the contribution guidelines.
- [ ] I have ensured my PR adheres to the project scope and contribution guidelines.
- [ ] All links in my PR are set to public access.

parthasarathydNU reviewed (changes requested):
Thanks for walking me through the changes @suprajakosuri01 

Further changes required:
1. Updating the readme of the repo - also verify if the readme of the project dashboard accurately reflects what the dashboard is about
2. Do more reading and internet research about the individual patterns to get better understanding
3. Refine the chatGpt conversations to explain the patterns better 
4. Split the slides into two - one for intro LLM and another for into to prompting patterns
5. Add comparison screenshots highlighting the difference between a basic conversation and conversation by using a prompt pattern in the slides

Let's aim to push these changes by next Monday. Good work on this PR!

suprajakosuri01 commented:
Updated readme of the repo 
Refined the chatGpt conversations to explain the patterns better
Added comparison screenshots highlighting the difference between a basic conversation and conversation by using a prompt pattern in the slides
Split the slides into two - one for intro LLM and another for into to prompting patterns

parthasarathydNU reviewed (changes requested):
Add the comparison screenshots to the PPT as discussed

For question refinement pattern, show examples of moving from a very vague to a very specific question

Refer this conversation for cognitive verifier pattern: https://chat.openai.com/share/d68b4afe-1b11-40de-9099-331991737f2f

Crop the images to omit whitespace around the chatgpt conversation screenshots

Add caption for all images, mentioning what is it that you want the students to see in the image

suprajakosuri01 commented:
1. Added the comparison screenshots to the PPT as discussed
2. Added question refinement pattern, showing examples of moving from a very vague to a very specific question
3. Added  cognitive verifier pattern
4. Croped the images to omit whitespace around the chatgpt conversation screenshots
5. Add caption for all images, mentioning what is it that i want the students to see in the image

parthasarathydNU reviewed (dismissed):
Amazing job on the sample conversations , they illustrate the concepts very well. Make sure to add the screenshots and examples to the PPT as well.

suprajakosuri01 commented:
Rebasing and sending another revision

parthasarathydNU reviewed (approved):
Congrats on completing your first project! Good job on the details, proper documentation through the readme files, clear direction in the case study and sufficient research and experimentation to illustrate the utility of prompting techniques. Looking forward to collaborating on the next!
